File: kraken/features/web/step_definitions/page_objects/tags_page.rb
class TagsPage
  attr_reader :driver

  def initialize(driver)
    @driver = driver
  end

  def go_to_create_new_tag
    @driver.find_element(
      :css, 'a[href="#/tags/new/"]'
    ).click
  end

  def click_tag_with_name(name)
    @driver.find_elements(
      :css, '.gh-tag-list-name'
    ).select do |element|
      element.text == name
    end.first.click
  end
end

File: playwright/PageObjects/TagsPage.js
module.exports = class TagsPage {
  constructor(page) {
    this.page = page;
  }

  async goToCreateNewTag() {
    return this.page.click('a[href="#/tags/new/"]');
  }

  async clickTagWithName(name) {
    return this.page.click(`.gh-tag-list-name:has-text("${name}")`);
  }

  async searchTagByName(nameTag) {
    await new Promise(r => setTimeout(r, 1000));
    const tagsTitles = await this.page.$$("h3.gh-tag-list-name");
    for (var i = 0; i < tagsTitles.length; i++) {
      if (await tagsTitles[i].innerText() === nameTag) {
        return true;
      }
    }
    return false;
  }

  async clickInternalTags() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('span:text("Internal tags")');
  }

  async clickPublicTags() {
    await new Promise(r => setTimeout(r, 1000));
    return this.page.click('span:text("Public tags")');
  }
};
